### DISCUSSION TITLE
Re-render a Feature when data changes

### DISCUSSION THREAD

**Discussion by jameslavery** [Mapsui/Mapsui#2734] [Q&A]
For context, displaying a Boat on a chart, at a given position and pointing in a known direction (its heading). I'm using a custom SkiaSharp renderer.

When the Boat's heading or position changes, I need to re-render the associated Feature(s). What's the most efficient way to do this? 

I assume I could call Map.RefreshGraphics(), but that sounds inefficient (unless it does internal cacheing). Is there a way to trigger only the Boat's Features to be re-rendered?

**Comment by pauldendulk** [comment]
In Mapsui we always redraw the entire screen. For the most important scenarios (panning, zooming, pinching) this is just necessary. 

If your viewport is static with just a few things moving it would be more efficient to only redraw what is changed. This would be important if you have many updates, or you need to limit energy usage. I can see why we would want this but I don't think this will be built soon, unless I specifically need it for some project.

**Comment by jameslavery** [comment]
Thank you, that is useful. I think performance is good enough to just redraw everything - we don't have that many artefacts to render.

James

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
